app/javascript/react/components/SessionFilters/ParameterFilter.tsx

```diff
@@ -23,34 +23,90 @@ import {
   SessionType,
   SessionTypes,
 } from "../../types/filters";
-import { useMapParams } from "../../utils/mapParamsHandler";
+import { Sensor } from "../../types/sensors";
+import { UserSettings } from "../../types/userStates";
+import { UrlParamsTypes, useMapParams } from "../../utils/mapParamsHandler";
+import { setSensor } from "../../utils/setSensor";
 import useMobileDetection from "../../utils/useScreenSizeDetection";
 import { CustomParameterFilter } from "./CustomParameterFilter";
 import { FilterInfoPopup } from "./FilterInfoPopup";
 import * as S from "./SessionFilters.style";
 
-const basicMeasurementTypes = (sessionType: SessionType) =>
+// Utility functions
+const getBasicMeasurementTypes = (sessionType: SessionType) =>
   sessionType === SessionTypes.FIXED
     ? FixedBasicParameterTypes
     : MobileBasicParameterTypes;
 
-interface ParameterFilterProps {
-  isBasicOpen: boolean;
-}
-
 export const filterCustomParameters = (
   parameters: string[],
-  sessionType: string
+  sessionType: SessionType
+) => {
+  const basicParameters = getBasicMeasurementTypes(sessionType);
+  return parameters.filter((param) => !basicParameters.includes(param));
+};
+
+// Reusable function to set parameters and URL parameters
+const setParameterParams = (
+  selectedParameter: ParameterType,
+  sensors: Sensor[],
+  sessionType: SessionType,
+  isMobile: boolean,
+  currentUserSettings: UserSettings,
+  isIndoor: string | null,
+  setUrlParams: (params: { key: UrlParamsTypes; value: string }[]) => void,
+  dispatch: (action: any) => void
 ) => {
-  const basicParameters =
-    sessionType === SessionTypes.FIXED
-      ? FixedBasicParameterTypes
-      : MobileBasicParameterTypes;
+  const sensorData = setSensor(selectedParameter, sensors, sessionType);
+  const commonParams = [
+    {
+      key: UrlParamsTypes.previousUserSettings,
+      value: currentUserSettings,
+    },
+    {
+      key: UrlParamsTypes.currentUserSettings,
+      value: isMobile
+        ? UserSettings.FiltersView
+        : currentUserSettings === UserSettings.CrowdMapView
+        ? UserSettings.CrowdMapView
+        : UserSettings.MapView,
+    },
+    {
+      key: UrlParamsTypes.measurementType,
+      value: selectedParameter,
+    },
+    {
+      key: UrlParamsTypes.sensorName,
+      value: sensorData.sensorName,
+    },
+    {
+      key: UrlParamsTypes.unitSymbol,
+      value: sensorData.unitSymbol,
+    },
+    {
+      key: UrlParamsTypes.currentZoom,
+      value: UrlParamsTypes.previousZoom,
+    },
+  ];
+
+  setUrlParams(commonParams);
 
-  return parameters.filter((param: string) => !basicParameters.includes(param));
+  if (isIndoor === "true" && sensorData.sensorName.startsWith("Gov")) {
```

Comment:
move string "true" to const. Maybe we can use `const ` for "Gov" that we already have in file `sensor.ts`